sdk/python/kfp/dsl/types/type_annotations_test.py

```diff
@@ -222,5 +224,89 @@ class NotArtifact:
         self.assertFalse(type_annotations.is_artifact_class(NotArtifact))
 
 
+class MyArtifact(dsl.Artifact):
+    pass
+
+
+class NotArtifact:
```

Comment:
We currently treat arbitrary unknown types as artifact types. This seems to be a change of behavior?

Reply:
For the purposes of this PR and these unit tests, I think this is not a behavior change. `issubclass_of_artifact(NotArtifact)` should be `True` as it does not subclass artifact.

Separately, I think the behavior of "falling back" to an artifact may no longer be the current state.

For example, this pipeline does not compile:

```python
@dsl.component
def comp(string: 'Artifact'):
    print(string)
```

Based on the error message, I think this was lost some time ago in https://github.com/kubeflow/pipelines/pull/8232.

Reply:
> `issubclass_of_artifact(NotArtifact)` should be `True` as it does not subclass artifact.

You meant `False`? More importantly, would `NotArtifact` compile into artifact or parameter? I think it's the former, right? If that's the case, I think there's an issue here.

In v1, the way to annotate an artifact would be

```python
@create_component_from_func
def comp(string: InputPath('Artifact')):
    with open(...) as f:
         print(f.read())
```
IIRC, while `Input/Output` accepts only subclasses of `dsl.Artifact`, `InputPath/OutputPath` should accepts everything with or without quotes. And we maintain this backward compatibility in v2. If this is no longer the current state, this seems to be an uncaught regression, and we should have a better understanding of the impact of this change. 

Maintaining v1 compatibility aside, it seems this behavior (whether it's new from this PR, or unintentionally changed in the past) is incompatible with our future plan of supporting common Python types (`pd.DataFrame` for example)? My impression was those types would be compiled into artifacts?

Reply:
Yes, thank you -- I meant `False`.

Neither of the following compiles successfully. Let's discuss this more synchronously.

```python
@dsl.component
def comp(string: dsl.InputPath('Artifact')):
    pass
```

```python
@dsl.component
def comp(string: dsl.InputPath(NotArtifact)):
    pass
```

Future looking, unless there's something I'm missing, I think that we are still able to add special support for `pd.DataFrame` (and similar) in the future, even if this is not currently supported.

Reply:
Synced offline, and this is now resolved after renaming the class name used in test.